File: mid/Array/128. Longest Consecutive Sequence (O(n)).cpp
// Time : O(n)
// Space : O(n)

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()<=1)
            return nums.size();
            
        sort(nums.begin(),nums.end());
        int c=1,mx=1;
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]-nums[i-1]==0)
                continue;
            else if(nums[i]-nums[i-1]==1)
            {
                c++;
                mx=max(mx,c);
            }
            else if(nums[i]-nums[i-1]>1)
            {
                mx=max(mx,c);
                c=1;
            }
        }
        return mx;
    }
};

File: easy/Hash Set/128. Longest Consecutive Sequence (O(n)).py
# https://leetcode.com/problems/longest-consecutive-sequence/solutions/6006950/simple-py-sol-beats-94/
# Time complexity: O(n)
# Space complexity: O(n) auxiliary space

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        nums_set = set(nums)
        res = 0
        
        while nums_set:
            curr = 1
            num = nums_set.pop()
            lower = num - 1
            higher = num + 1
            while lower in nums_set:
                nums_set.remove(lower)
                lower -= 1
                curr += 1
            while higher in nums_set:
                nums_set.remove(higher)
                higher += 1
                curr += 1
            res = max(res, curr)

        return res